### PR TITLE
Repo

### PR DESCRIPTION
**Pull request by ne-grebe**

### CODE REVIEW

File: src/Repositories/TodoRepository.ts [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+}
+
+//interface to implement repository pattern
+export interface RepositoryPI<T> {
+    add(objectToAdd: T);
+    getAll();
+    getById(objectID: string);
+    getByCompletionStatus(completedStatus: boolean);
+    remove(objectToDelete: string);
+    update(objectToChange: string);
```
Comment (lawhorkl) [inline src/Repositories/TodoRepository.ts]: Shouldnt the parameter here be type `T`?

File: src/Repositories/TodoRepository.ts [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    completedStatus: boolean
+}
+
+//interface to implement repository pattern
+export interface RepositoryPI<T> {
+    add(objectToAdd: T);
+    getAll();
+    getById(objectID: string);
+    getByCompletionStatus(completedStatus: boolean);
+    remove(objectToDelete: string);
```
Comment (lawhorkl) [inline src/Repositories/TodoRepository.ts]: Why give the remove function the entire object? doesnt it just need the ID?

File: src/Repositories/TodoRepository.ts [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+        * as task array grows to dozens or hundreds of tasks, may want to consider how else to
+        * remove a specific element without having to create new array
+        **/
+
+        const filteredTodos = this.taskArray.filter(todo => todo.id === todoID)
+        const todoToBeDeleted = filteredTodos[0]
+        const position = filteredTodos.indexOf(todoToBeDeleted)
+
+        this.taskArray.splice(position, 1)
+        return todoToBeDeleted
```
Comment (lawhorkl) [inline src/Repositories/TodoRepository.ts]: Why return the deleted object?

File: src/Repositories/TodoRepository.ts [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+    public update(todoID: string) {
+        /**
+             * Currently implementing hard coded update for 'Marrying the queen' task
+             * But if id is changed, will update whatever task that was, regardless
+             **/
+         const requestedTask = this.taskArray.filter(x => x.id === todoID);
+
+         requestedTask[0].name = 'Divorce the queen'
+         requestedTask[0].dueDate = (new Date(2023, 6, 5, 18, 0, 0)).toString()
+         requestedTask[0].completedStatus = true
```
Comment (lawhorkl) [inline src/Repositories/TodoRepository.ts]: Why are you hard coding this? The update function should probably take an `id` parameter and a `T` parameter, you should not need to hard code anything.

File: src/Routes/todoRoutes.ts [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+        handler: (request: Request, h: ResponseToolkit) => {
+            var payload = request.payload
+            console.log(payload)
+            const todo: Todo = {
+                id: uuidv4(),
+                name: payload['name'],
+                createdAt: new Date(),
+                dueDate: payload['dueDate'],
+                completedStatus: false
+            }
```
Comment (lawhorkl) [inline src/Routes/todoRoutes.ts]: golden opportunity for the spread operator

**Review by lawhorkl — CHANGES_REQUESTED** [review]
Overall I think you are really close, excellent work!

File: src/Repositories/TodoRepository.ts (hunk 2 above)
Comment (ne-grebe) [reply src/Repositories/TodoRepository.ts]: you're right; it was only taking in the ID already, so I changed "objectToDelete" to "objectID"

File: src/Repositories/TodoRepository.ts (hunk 1 above)
Comment (ne-grebe) [reply src/Repositories/TodoRepository.ts]: Changed to an id to get the task that will be updated and another of type T to pass the payload through

File: src/Repositories/TodoRepository.ts (hunk 3 above)
Comment (ne-grebe) [reply src/Repositories/TodoRepository.ts]: Now returning remaining array

File: src/Repositories/TodoRepository.ts [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+    public update(todoID: string, updatePayload: Todo) {
+        /**
+             * Currently implementing hard coded update for 'Marrying the queen' task
+             * But if id is changed, will update whatever task that was, regardless
+             **/
+        const requestedTask = this.taskArray.filter(x => x.id === todoID);
+
+        let updatedTask: Todo = {
+            ...requestedTask[0], ...updatePayload
+        }
```
Comment (lawhorkl) [inline src/Repositories/TodoRepository.ts:82]: NICE!!!

**Review by lawhorkl — APPROVED** [review]

**Pull request merged**